### PR TITLE
tls_server add linux for marking the socket

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/cli/tls_server.cpp [hunk 1]
```diff
@@ -16,7 +16,14 @@
...
-#if defined(SO_USER_COOKIE) || defined(SO_RTABLE)
+#if defined(SO_MARK) || defined(SO_USER_COOKIE) || defined(SO_RTABLE)
+#if defined(SO_MARK)
+#define BOTAN_SO_SOCKETID SO_MARK
+#elif defined(SO_USER_COOKIE)
+#define BOTAN_SO_SOCKETID SO_USER_COOKIE
+#else
+#define BOTAN_SO_SOCKETID SO_RTABLE
+#endif
 #define SOCKET_ID 1
```
Comment (randombit) [inline src/cli/tls_server.cpp:27]: Can we remove the definition of `SOCKET_ID` and just use the existence of `BOTAN_SO_SOCKETID` to decided if we should call `setsockopt` or not?

**Review by randombit — COMMENTED** [review]
Question / possible change request

File: src/cli/tls_server.cpp (hunk 1 above)
Comment (devnexen) [reply src/cli/tls_server.cpp]: valid point

Comment (randombit) [reply src/cli/tls_server.cpp]: Looks good, thanks!

**Pull request merged**